x-pack/test/performance/journeys/promotion_tracking_dashboard/promotion_tracking_dashboard.ts

```diff
@@ -48,16 +49,7 @@ export default function ({ getService }: FtrProviderContext) {
           {
             name: 'Wait for visualization animations to finish',
             handler: async ({ page }) => {
-              await page.waitForFunction(function renderCompleted() {
-                const visualizations = Array.from(
-                  document.querySelectorAll('[data-rendering-count]')
-                );
-                const visualizationElementsLoaded = visualizations.length > 0;
-                const visualizationAnimationsFinished = visualizations.every(
-                  (e) => e.getAttribute('data-render-complete') === 'true'
-                );
-                return visualizationElementsLoaded && visualizationAnimationsFinished;
-              });
+              await waitForVisualizations(page, 1);
```

Comment:
the events still need to reach the server - the loading happens in the browser so the browser needs to "send" the event to the server :)

Reply:
Even though I believe the latency is really low and most probably we won't have so many dropped events it is still possible as you mentioned, so I agree 👍🏽. However not really sure if we have mechanism to ensure all events are sent successfully and it's ready shutdown, if we don't have some check for that we will probably end up adding arbitrary delay at the end of journeys. 

Anyone has an idea on this ? cc: @spalger , @afharo

Reply:
I expect that when a step completes we can wait for both 3 seconds and for any requests to the telemetry endpoint to complete. If no requests to the telemetry endpoint are started in the 3 second period then telemetry was sent before the step completed. I'd be happy to help you get this logic implemented quickly @suchcodemuchwow. @afharo can you tell us how we can identify which requests are "telementry shipping" requests? What hostname pattern should we look for?

Reply:
They look like this:
```
:authority: telemetry-staging.elastic.co
:method: POST
:path: /v3/send/kibana-browser
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: en-GB,en-US;q=0.9,en;q=0.8,de;q=0.7
content-length: 1103
content-type: application/x-ndjson
origin: http://localhost:5601
referer: http://localhost:5601/app/dashboards
sec-ch-ua: "Google Chrome";v="105", "Not)A;Brand";v="8", "Chromium";v="105"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "macOS"
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: cross-site
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36
x-elastic-cluster-id: FUf8kO2WR9uDG1ehZcC88A
x-elastic-license-id: 77941f60-b2dc-46d2-b9a0-0767b5a73a3e
x-elastic-stack-version: 8.5.0
```

Reply:
It's a good call to wait for them - they tend to be relatively slow because they wait for the events to be persisted in Elasticsearch:
<img width="389" alt="Screenshot 2022-09-08 at 11 28 26" src="https://user-images.githubusercontent.com/1508364/189087302-3651ee24-2e41-4c31-b92f-c07b65e5922e.png">